### PR TITLE
Implemented adl handler contract functions with additional functions beyond issue requirements; ready for rollback if needed.

### PR DESCRIPTION
**Pull request by ametel01**
## Pull Request type
- Feature

Resolves: #275 

## Other information
It appears I may have misunderstood the original scope of the issue. I've implemented not only the functions directly related to the issue but also additional ones in the same modules. If these extra implementations are not needed, I can revert those changes and only include the features directly related to the issue at hand.

Note: Comments in the code are still a work in progress and will be refactored and completed.

### CODE REVIEW

**Comment by sparqet** [comment]
> ## Pull Request type
> * Feature
> 
> Resolves: #275
> 
> ## Other information
> It appears I may have misunderstood the original scope of the issue. I've implemented not only the functions directly related to the issue but also additional ones in the same modules. If these extra implementations are not needed, I can revert those changes and only include the features directly related to the issue at hand.
> 
> Note: Comments in the code are still a work in progress and will be refactored and completed.

No worries mate, can you revert the extra functions you implemented please, because there is already people working on it  😁

File: src/exchange/adl_handler.cairo [hunk 1]
```diff
@@ -56,40 +59,48 @@ mod AdlHandler {
...
 
     /// ExecuteAdlCache struct used in execute_adl.
-    #[derive(Drop, starknet::Store, Serde)]
+    #[derive(Drop, Serde)]
     struct ExecuteAdlCache {
         /// The starting gas to execute adl.
         starting_gas: u128,
         /// The min oracles block numbers.
-        min_oracle_block_numbers: Array<u64>,
+        min_oracle_block_numbers: Array<u128>,
```
Comment (Sk8erboi84) [inline src/exchange/adl_handler.cairo:101]: You need to keep u64

File: src/exchange/adl_handler.cairo [hunk 2]
```diff
@@ -56,40 +59,48 @@ mod AdlHandler {
...
         /// The max oracles block numbers.
-        max_oracle_block_numbers: Array<u64>,
```
Comment (Sk8erboi84) [inline src/exchange/adl_handler.cairo:92]: You need to keep u64

**Comment by ametel01** [comment]
closing this and opening an new clean PR

**Comment by ametel01** [comment]
closing this and opening an new clean PR

**Pull request closed without merging**